src/hooks/useModal/useModal.test.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+
+import { Button, Modal } from '../..'
+import { useModal } from '.'
+
+describe('useModal', () => {
```

Comment:
I would also add unit tests for this hook:

```suggestion
describe('useModal', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('returns isModalVisible, openModal, closeModal and toggleModal properties', async () => {
    const {result} = renderHook(() => useModal())

    const {isModalVisible, openModal, closeModal, toggleModal} = result.current
    expect(isModalVisible).toEqual(false)

    openModal()
    expect(result.current.isModalVisible).toEqual(true)

    closeModal()
    expect(result.current.isModalVisible).toEqual(false)

    toggleModal()
    expect(result.current.isModalVisible).toEqual(true)

    toggleModal()
    waitFor(() => expect(result.current.isModalVisible).toEqual(false))
  })
```